internal/credential/service_list_libraries.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package credential
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/pagination"
+)
+
+// LibraryService defines the interface expected
+// to list, estimate the count of and list deleted items of
+// credential libraries.
+type LibraryService interface {
+	EstimatedCount(context.Context) (int, error)
+	ListDeletedIds(context.Context, time.Time) ([]string, time.Time, error)
+	List(context.Context, string, ...Option) ([]Library, error)
+}
```

Comment:
The library service is implemented by any credential library repos, in this case just the vault repo.